internal/packages/installer/zip_installer.go

```diff
@@ -0,0 +1,67 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package installer
+
+import (
+	"fmt"
+
+	"github.com/Masterminds/semver/v3"
+	"github.com/pkg/errors"
+
+	"github.com/elastic/elastic-package/internal/kibana"
+	"github.com/elastic/elastic-package/internal/packages"
+)
+
+var semver8_7_0 = semver.MustParse("8.7.0")
+
+type zipInstaller struct {
+	zipPath  string
+	manifest packages.PackageManifest
+
+	kibanaClient *kibana.Client
+}
+
+// CreateForZip function creates a new instance of the installer.
+func CreateForZip(zipPath string, m packages.PackageManifest) (Installer, error) {
+	kibanaClient, err := kibana.NewClient()
+	if err != nil {
+		return nil, errors.Wrap(err, "could not create kibana client")
+	}
+	kibanaVersion, err := kibanaClient.Version()
+	if err != nil {
+		return nil, err
+	}
+	v, err := semver.NewVersion(kibanaVersion.Number)
+	if err != nil {
+		return nil, fmt.Errorf("invalid Kibana version")
+	}
+	if v.LessThan(semver8_7_0) {
+		return nil, fmt.Errorf("not supported uploading zip packages in Kibana %s", kibanaVersion.Number)
+	}
```

Comment:
Probably, this check about the Kibana version needs to be moved into `cmd/install.go`. This would be needed to check the version to start using this method by default when running `elastic-package install` in a package directory in case Kibana >= 8.7.0.

Reply:
I wanted to avoid changing the signature of this method (or CreateForManifest) since it is used in testrunners too (assets). But I could take a look at it.